server.js

```diff
@@ -66,9 +66,15 @@ app.use(api);
 
 function hasValidEmail(req) {
   const emails = req.user.emails
-    .map((email) => email.value)
-    .join('');
-  return emails.indexOf('@zenika') != -1 || emails.indexOf('zenika.resume@gmail.com') != -1;
+    .map((email) => email.value);
+  const validEmails = emails.filter(email => (email.indexOf('@zenika.com') !== -1 || email.indexOf('zenika.resume@gmail.com') !== -1));
+  if (validEmails.length >= 1) {
+    if (validEmails.length === 1 && validEmails[0].indexOf('-ext@') !== -1) {
+      return false;
+    }
+    return true;
+  }
+  return false;
```

Comment:
You may use `Array.prototype.some` to convey to the reader the "at least one" part of the algorithm.